### PR TITLE
Chore/refactor rollup

### PR DESCRIPTION
**Pull request by lifeiscontent**
@edmundhung I'm not sure why, but rollup will not build within the packages directory, do you have any ideas? (I've just converted 1 package for testing purposes)

### CODE REVIEW

**Comment by edmundhung** [comment]
Maybe the copy plugin is failed because it couldn't find the readme and license files?

**Comment by lifeiscontent** [comment]
@edmundhung I just tried to update copy plugin locally, but the error has to do with the TS code.

```
 ./index.ts → ....
│ [!] (plugin babel) SyntaxError: /Users/lifeiscontent/GitHub/edmundhung/conform/packages/conform-dom/index.ts: Unexpect…
│   1 | export {
│ > 2 |  Otype Combine,
│     |  t     ^
│   3 |  ttype Constraint,
│   4 |  Mtype ControlButtonProps,
│   5 | ngtype FormId,
│ index.ts (2:6)
│ SyntaxError: /Users/lifeiscontent/GitHub/edmundhung/conform/packages/conform-dom/index.ts: Unexpected token, expected …
│   1 | export {
│ > 2 |   type Combine,
│     |        ^
│   3 |   type Constraint,
│   4 |   type ControlButtonProps,
│   5 |   type FormId,
│     at toParseError (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7.23.5/node_modul…
│     at Parser.raise (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7.23.5/node_modul…
│     at Parser.unexpected (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7.23.5/node_…
│     at Parser.expect (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7.23.5/node_modu…
│     at Parser.parseExportSpecifiers (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7…
│     at Parser.maybeParseExportNamedSpecifiers (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babe…
│     at Parser.parseExport (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7.23.5/node…
│     at Parser.parseStatementContent (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7…
│     at Parser.parseStatementLike (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7.23…
│     at Parser.parseModuleItem (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7.23.5/…
│     at Parser.parseBlockOrModuleBlockBody (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+pa…
│     at Parser.parseBlockBody (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7.23.5/n…
│     at Parser.parseProgram (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7.23.5/nod…
│     at Parser.parseTopLevel (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7.23.5/no…
│     at Parser.parse (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7.23.5/node_modul…
│     at parse (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+parser@7.23.5/node_modules/@bab…
│     at parser (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+core@7.23.5/node_modules/@babe…
│     at parser.next (<anonymous>)
│     at normalizeFile (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+core@7.23.5/node_module…
│     at normalizeFile.next (<anonymous>)
│     at run (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+core@7.23.5/node_modules/@babel/c…
│     at run.next (<anonymous>)
│     at transform (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+core@7.23.5/node_modules/@b…
│     at transform.next (<anonymous>)
│     at step (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensy…
│     at evaluateAsync (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modu…
│     at /Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync/ind…
│     at new Promise (<anonymous>)
│     at fn (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync…
│     at stopHiding - secret - don't use this - v1 (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@b…
│     at Object.transformAsync (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@babel+core@7.23.5/nod…
│     at transformCode (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/@rollup+plugin-babel@5.3.1_@ba…
│     at transform (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/rollup@2.79.1/node_modules/rollup/…
│     at ModuleLoader.addModuleSource (/Users/lifeiscontent/GitHub/edmundhung/conform/node_modules/.pnpm/rollup@2.79.1/n…
└
```

**Comment by lifeiscontent** [comment]
figured it out, I'd forgotten to copy the babel config to the CJS config.

@edmundhung if this looks good to you, I'll go ahead and migrate all the other packages.

File: packages/conform-dom/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+				exclude: /node_modules/,
+				extensions: ['.ts', '.tsx'],
+			}),
+			nodeResolve({
+				extensions: ['.ts', '.tsx'],
+			}),
+			copy({
+				targets: [
+					{ src: `../../README`, dest: sourceDir },
+					{ src: `../../LICENSE`, dest: sourceDir },
```
Comment (lifeiscontent) [inline packages/conform-dom/rollup.config.js:51]: let me know if you want to handle this differently

**Review by edmundhung — COMMENTED** [review]
Looks good to me

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -2,63 +2,60 @@ name: Tests
...
-        node-version: 20
-    - uses: pnpm/action-setup@v2
-      with:
-        version: 8
-    - name: Download deps
-      run: pnpm i
-    - name: Install Playwright
-      run: pnpm exec playwright install --with-deps
-    - name: Build packages
-      run: pnpm run build
```
Comment (lifeiscontent) [inline .github/workflows/tests.yml:55]: this is the only thing that changed, the rest was format issues with prettier I believe.

File: packages/conform-dom/package.json [hunk 3]
```diff
@@ -38,5 +47,14 @@
 		"validation",
 		"dom"
 	],
-	"sideEffects": false
+	"sideEffects": false,
+	"devDependencies": {
+		"@babel/core": "^7.17.8",
+		"@babel/preset-env": "^7.20.2",
+		"@babel/preset-react": "^7.18.6",
```
Comment (edmundhung) [inline packages/conform-dom/package.json:54]: This dependency can be removed.

File: packages/conform-yup/package.json [hunk 4]
```diff
@@ -29,6 +38,14 @@
 		"yup": ">=0.32.0"
 	},
 	"devDependencies": {
+		"@babel/core": "^7.17.8",
+		"@babel/preset-env": "^7.20.2",
+		"@babel/preset-react": "^7.18.6",
```
Comment (edmundhung) [inline packages/conform-yup/package.json:43]: This one too.

File: packages/conform-zod/package.json [hunk 5]
```diff
@@ -31,6 +40,14 @@
 		"zod": "^3.21.0"
 	},
 	"devDependencies": {
+		"@babel/core": "^7.17.8",
+		"@babel/preset-env": "^7.20.2",
+		"@babel/preset-react": "^7.18.6",
```
Comment (edmundhung) [inline packages/conform-zod/package.json:45]: Same as above

**Review by edmundhung — APPROVED** [review]
This looks great! Just a few minor issues

**Comment by lifeiscontent** [comment]
@edmundhung updated :)

**Review by edmundhung — APPROVED** [review]
Let's get it merged!

**Pull request merged**